core/src/main/scala/scalaz/MonadRecSafely.scala

```diff
@@ -0,0 +1,34 @@
+package scalaz
+
+import Isomorphism._
+
+/**
+  * Takes functions operating on Bind and converts them into a BindRec version.
+  *
+  * Port of Phil Freeman's work in PureScript:
+  *
+  * https://github.com/paf31/purescript-safely/blob/ef7377d7f7659250f2f4a5da02c5d229e4eeed53/src/Control/Safely.purs#L19
+  */
+trait MonadRecSafely[M[_], A] {
+  def apply[S[_]: BindRec: Applicative](lift: M <~> S): S[A]
```

Comment:
Seems like one-directional `lift: M ~> S` would be sufficient.

Reply:
I agree with @TomasMikula - we need at least a `M ~> S` and `S ~> M` here. I agree this doesn't necessarily imply isomorphism, which is too strong of a condition I think. Just an adjunction will do.